core/embed/rust/src/ui/model_mercury/flow/continue_recovery.rs

```diff
@@ -0,0 +1,235 @@
+            ComponentExt, SwipeDirection,
+        },
+        flow::{
+            base::{DecisionBuilder as _, StateChange},
+            FlowMsg, FlowState, SwipeFlow,
+        },
+        layout::obj::LayoutObj,
+        model_mercury::component::{CancelInfoConfirmMsg, PromptScreen, SwipeContent},
+    },
+};
+
+use super::super::{
+    component::{Frame, FrameMsg, VerticalMenu, VerticalMenuChoiceMsg},
+    theme,
+};
+
+const RECOVERY_TYPE_NORMAL: u32 = 0;
+const RECOVERY_TYPE_DRY_RUN: u32 = 1;
+const RECOVERY_TYPE_UNLOCK_REPEATED_BACKUP: u32 = 2;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ContinueRecoveryBeforeShares {
+    Main,
+    Menu,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ContinueRecoveryBetweenShares {
+    Main,
+    Menu,
+    CancelIntro,
+    CancelConfirm,
+}
+
+impl FlowState for ContinueRecoveryBeforeShares {
+    #[inline]
+    fn index(&'static self) -> usize {
+        *self as usize
+    }
+
+    fn handle_swipe(&'static self, direction: SwipeDirection) -> StateChange {
+        match (self, direction) {
+            (Self::Main, SwipeDirection::Left) => Self::Menu.swipe(direction),
+            (Self::Menu, SwipeDirection::Right) => Self::Main.swipe(direction),
+            (Self::Main, SwipeDirection::Up) => self.return_msg(FlowMsg::Confirmed),
+            _ => self.do_nothing(),
+        }
+    }
+
+    fn handle_event(&'static self, msg: FlowMsg) -> StateChange {
+        match (self, msg) {
+            (Self::Main, FlowMsg::Info) => Self::Menu.transit(),
+            (Self::Menu, FlowMsg::Cancelled) => Self::Main.swipe_right(),
+            (Self::Menu, FlowMsg::Choice(0)) => self.return_msg(FlowMsg::Cancelled),
+            _ => self.do_nothing(),
+        }
+    }
+}
+
+impl FlowState for ContinueRecoveryBetweenShares {
+    #[inline]
+    fn index(&'static self) -> usize {
+        *self as usize
+    }
+
+    fn handle_swipe(&'static self, direction: SwipeDirection) -> StateChange {
+        match (self, direction) {
+            (Self::Main, SwipeDirection::Left) => Self::Menu.swipe(direction),
+            (Self::Menu, SwipeDirection::Right) => Self::Main.swipe(direction),
+            (Self::Main, SwipeDirection::Up) => self.return_msg(FlowMsg::Confirmed),
+            (Self::CancelIntro, SwipeDirection::Up) => Self::CancelConfirm.swipe(direction),
+            (Self::CancelIntro, SwipeDirection::Right) => Self::Menu.swipe(direction),
+            (Self::CancelConfirm, SwipeDirection::Down) => Self::CancelIntro.swipe(direction),
+            _ => self.do_nothing(),
+        }
+    }
+
+    fn handle_event(&'static self, msg: FlowMsg) -> StateChange {
+        match (self, msg) {
+            (Self::Main, FlowMsg::Info) => Self::Menu.transit(),
+            (Self::Menu, FlowMsg::Choice(0)) => Self::CancelIntro.swipe_left(),
+            (Self::Menu, FlowMsg::Cancelled) => Self::Main.swipe_right(),
```

Comment:
Yes, it will send BR too.

Yes, I think Suite should be fine with it because sending BRs behaves in the same way as in the old implementation of `homescreen_dialog`, i.e. the success/instruction screen sends BR and the confirmation of flow cancellation also sends BR. These two screens are called in a loop (until you confirm the success/instruction screen) so they possibly send the BRs repeatedly. Anyway, I'll ask QA for thorough testing of these scenarios.